### ISSUE TITLE
MiniMagick Filling up /tmp/ directory.

### ISSUE THREAD

**Issue by scande3** [samvera/hydra-derivatives#40]
I was running derivative creation on the ~75k objects in our repository. However, suddenly, all of my derivatives started error-ing and the AVI Processing server reported it had no space left on its hard drive. Looking at /tmp/, there are tons of files like: mini_magick20140730-1251-4a7xzf.jpg that ate up all of the available space (~60 GB of them).

I recalled that when using MiniMagick in the past, I was using a specific unmerged pull request from someone else for that exact issue in the past (though I stuck with RMagick in the end due to other image processing bugs MiniMagick has). Regardless, the relevant unmerged pull request is: https://github.com/minimagick/minimagick/pull/188

Once the current backlog of ~2k remaining failed objects finishes, I'll attempt to use that old pull request to confirm it fixes the issue as it did in the past. Assuming it does, not sure if the best course of action is to pressure MiniMagick about accepting it to fix there bug or just using that unmerged version of the code?

To duplicate the jpg spam, just create a thumbnail derivative from a tif and check your /tmp/ directory. These files will get deleted upon a restart (by default) but MiniMagick leaving them behind when done with them just doesn't seem like correct behavior.

**Comment by eocarragain** [comment]
We're still encountering this issue using minimagick 4.2.10. We rely on a cron script to delete the temp files

**Comment by narogers** [comment]
Are the people running into problems using Resque to process jobs in the background? I had them same problem for a while until I added RUN_AT_EXIT_HOOKS to my configuration. There is more information at https://github.com/resque/resque/issues/1167.

**Comment by mjgiarlo** [comment]
Yes, and we document this pattern in the Sufia README as well. (Thanks for that, @narogers!)
